### PR TITLE
[feat]: [IDP-306]: Sync app-config

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46433)

### CODE REVIEW

**Review by vikyathharekal — CHANGES_REQUESTED** [review]
> Reviewed 8 of 10 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46433)* status: 10 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`idp-service/config/config.yml` line 94 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTI-JPhABzA48wkWXu6:-NTI-JPi0uzWzeFig5WD:b-xzjmt1) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/config/config.yml#L94)):*
> ```YAML
> 
> backstageHttpClientConfig:
>   baseUrl: http://localhost:7007/api/
> ```

Can we use this URL itself?



_Code quote:_
```YAML
baseUrl
```
___
*[`idp-service/config/config.yml` line 143 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTHyd6WCqw_6A0_YFFp:-NTHyd6WCqw_6A0_YFFq:bpl3zr0) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/config/config.yml#L143)):*
> ```YAML
> prEnvDefaultBackstageNamespace: ""
> 
> appBaseUrl: localhost:3000
> ```

can we add backstage in the name somewhere?
___
*[`idp-service/src/main/resources/app-config.yml` line 70 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTI-dK8-joanHkYhxfE:-NTI-dK8-joanHkYhxfF:betsu9r) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/src/main/resources/app-config.yml#L70)):*
> ```YAML
> 
> harnessDeployment:
>   accountIdentifier: ${HARNESS_ACCOUNT_IDENTIFIER}
> ```

Do we still need these? cc: @SarvJ1  



_Code quote:_
```YAML
accountIdentifier
```

**Comment by sarthakkasat** [comment]
*[`idp-service/config/config.yml` line 94 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTI-JPhABzA48wkWXu6:-NTNv3ZKDVynHBQ-aBgh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/config/config.yml#L94)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Can we use this URL itself?
</blockquote></details>

Done.

**Comment by sarthakkasat** [comment]
*[`idp-service/config/config.yml` line 143 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTHyd6WCqw_6A0_YFFp:-NTNv6F71XgE4iPLoufD:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/config/config.yml#L143)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

can we add backstage in the name somewhere?
</blockquote></details>

Done.

**Comment by sarthakkasat** [comment]
*[`idp-service/src/main/resources/app-config.yml` line 70 at r2](https://reviewable.io/reviews/harness/harness-core/46433#-NTI-dK8-joanHkYhxfE:-NTNvC_062qWiqh9OtW4:b-xiz8ox) ([raw file](https://github.com/harness/harness-core/blob/e05f88bc82081d214dc3fb2775f23c21b309c9cf/idp-service/src/main/resources/app-config.yml#L70)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Do we still need these? cc: @SarvJ1  
</blockquote></details>

Checked from harness-idp it is removed

**Comment by sarthakkasat** [comment]
trigger utAll

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 6 of 7 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46433)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Comment by sarthakkasat** [comment]
trigger ut2

**Review by devesh-harness — DISMISSED** [review]

**Review by vikyathharekal — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46433#-:-NTRXVRyEScE-N7W4eDt:bnfp4nl)* status: 10 file(s) to review.

**Comment by vikyathharekal** [comment]
trigger review

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -662,6 +662,26 @@ paths:
...
+      description: API to sync app-config for a given account. This API can be accessed by Support User only.
+      x-internal: true
+      security:
+        - x-api-key: []
+      parameters:
+        - $ref: '#/components/parameters/AccountHeader'
+        - schema:
+            type: boolean
+          in: query
+          name: base-config
```
Comment (mankrit-singh) [inline idp-service/contracts/openapi/v1/openapi.yaml:681]: nit, this should be base_config

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 2 of 10 files at r1, 1 of 2 files at r2, 5 of 7 files at r3, 1 of 1 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46433)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46433)* status: :shipit: complete! ✓

**Comment by vikyathharekal** [comment]
trigger review

**Review by devesh-harness — APPROVED** [review]

**Pull request merged**